src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -75,6 +75,14 @@ namespace session {
 
 namespace {
 
+// TODO: asyncRpc - save_document_diff will call isWebsiteProject() that
+// needs to check the R runtime for whether Rmarkdown is installed. This is
+// not entirely accurate since if the first call to isWebsiteProject() is made
+// offline, it will return the wrong thing. Can we just cache a flag globally
+// whether RMarkdown is available - compute at startup and update when new libs
+// are added or will that state ever change?
+bool s_lastWebsiteProject;
```

Comment:
I believe that once RStudio has finished initialization, it is not possible to change the project type within that same session without restarting RStudio (the front-end, not just the rsession itself). Given that, I think it would be reasonable to cache the project type on startup, and then read or use that as appropriate.